syntax = "proto3";

package server_services;

// Request message for sending a file
message SendFileRequestBody {
  bytes file = 1;  // File content as bytes
  string file_name = 2;  // File name (optional)
}

// Response message for sending a file
message SendFileResponseBody {
  bool success = 1;  // Indicates whether the operation was successful
  string message = 2;  // Status message
}

// Request message for sending file chunks
message SendFileChunksRequest {
  bytes data = 1;  // Chunk of the file
  string file_name = 2;  // File name (optional, sent in the first chunk or metadata)
}

// Response message for sending file chunks
message SendFileChunksResponse {
  bool success = 1;  // Indicates if the upload was successful
  string message = 2;  // Additional status message
}

// Service definition for file sending operations
service SendFileService {
  // Unary RPC to send a file
  rpc SendFile (SendFileRequestBody) returns (SendFileResponseBody);

  // Streaming RPC to send file chunks
  rpc SendFileChunks (stream SendFileChunksRequest) returns (SendFileChunksResponse);
}